rustls/src/crypto/ring/ticketer.rs

```diff
@@ -194,4 +194,27 @@ mod tests {
         assert_eq!(t.decrypt(&cipher2).unwrap(), b"ticket 2");
         assert_eq!(t.decrypt(&cipher3).unwrap(), b"ticket 3");
     }
+
+    #[test]
+    fn aeadticketer_is_debug_and_producestickets() {
+        use super::*;
+        use alloc::format;
+
+        let key = [0u8; 32];
+        let alg = &aead::CHACHA20_POLY1305;
+        let key = aead::UnboundKey::new(alg, &key).unwrap();
+
+        let t = AeadTicketer {
+            alg,
+            key: aead::LessSafeKey::new(key),
+            lifetime: 1234,
+        };
+
+        assert_eq!(
+            format!("{:?}", t),
+            "AeadTicketer { alg: CHACHA20_POLY1305, lifetime: 1234 }"
+        );
+        assert!(t.enabled());
+        assert_eq!(t.lifetime(), 1234);
+    }
```

Comment:
I think this test can be simplified if you use `make_ticket_generator()` - was there a reason you wanted to avoid that?

```rust
#[test]
    fn aeadticketer_is_debug_and_producestickets() {
        use super::*;
        use alloc::format;

        let t = make_ticket_generator().unwrap();

        assert_eq!(
            format!("{:?}", t),
            "AeadTicketer { alg: CHACHA20_POLY1305, lifetime: 43200 }"
        );
        assert!(t.enabled());
        assert_eq!(t.lifetime(), 43200);
    }
```

(_Note: had to change the expected lifetime - but the value used is static with the impl_)

Reply:
oh, good idea!